### PR TITLE
Add RateLimitTransport and its documentation

### PR DESCRIPTION
**Pull request by juherr**
As requested in #30, I propose a documentation update to manage the API rate limit.

I added the implementation as optional too.

Feel free to edit the change :)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,7 @@
 *.sqlite
 *.done
 composer.lock
+composer.phar
```
Comment (frenchcomp) [inline .gitignore]: This is must be in your global gitignore and not in the project's gitignore

File: src/Transport/RateLimitTransport.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<?php
```
Comment (frenchcomp) [inline src/Transport/RateLimitTransport.php:1]: If it is a part of documentation, this class must be in documentation folder.
Else tests with PHPUnit are mandatories

File: src/Transport/RateLimitTransport.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    public function createUri(string $uri = ''): UriInterface
+    {
+        return $this->transport->createUri($uri);
+    }
+
+    public function asyncExecute(RequestInterface $request): PromiseInterface
+    {
+        $this->cnt++;
+        if ($this->cnt === $this->rateLimit) {
+            $stop = microtime(true);
```
Comment (frenchcomp) [inline src/Transport/RateLimitTransport.php:76]: PHP's base functions must be add below `use` class with `use function`

File: src/Transport/RateLimitTransport.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+declare(strict_types=1);
+
+namespace Teknoo\Sellsy\Transport;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UriInterface;
+
+class RateLimitTransport implements TransportInterface
```
Comment (frenchcomp) [inline src/Transport/RateLimitTransport.php:38]: Specify that this class has a blocking behavior because it use sleep

**Comment by frenchcomp** [comment]
Thanks, I had write some comments in my review. Tests are really mandatory.
You're welcome.

**Comment by juherr** [comment]
@fdglefevre I don't see any comment: I think you didn't finish the review.
Any idea about how to test it?

**Review by frenchcomp — CHANGES_REQUESTED** [review]

**Comment by frenchcomp** [comment]
Sorry, I lost to finish the review :/

File: src/Transport/RateLimitTransport.php (hunk 2 above)
Comment (juherr) [reply src/Transport/RateLimitTransport.php:1]: Tests done

File: src/Transport/RateLimitTransport.php (hunk 4 above)
Comment (juherr) [reply src/Transport/RateLimitTransport.php:38]: Done

File: src/Transport/RateLimitTransport.php (hunk 3 above)
Comment (juherr) [reply src/Transport/RateLimitTransport.php:76]: Done

**Comment by frenchcomp** [comment]
It's merged. Thanks you

**Pull request merged**